app/ts/components/pages/SettingsView.tsx

```diff
@@ -190,39 +189,39 @@ export function SettingsView() {
 	</main>
 }
 
-const RpcList = ({ rpcEntries }: { rpcEntries: RpcEntries }) => {
-	const [lastEntry] = rpcEntries
+const RpcListings = () => {
+	const rpcEntries = useRpcConnectionsList()
+	const latestEntry = useComputed(() => rpcEntries.value.at(0))
 
-	const loadDefaultRpcs = () => {
-		sendPopupMessageToBackgroundPage({
-			method: 'popup_set_rpc_list',
-			data: defaultRpcs
-		})
-	}
+	const loadDefaultRpcs = () => sendPopupMessageToBackgroundPage({ method: 'popup_set_rpc_list', data: defaultRpcs })
 
-	if (rpcEntries.length < 2 && lastEntry) {
+	if (rpcEntries.value.length < 2 && latestEntry.value !== undefined) {
 		return (
 			<>
 				<aside class = 'report' style = { { display: 'grid', height: '9rem', textAlign: 'center', rowGap: '0.5rem'} }>
 					<p style = { { color: '#ffffff80' } }>Interceptor requires at least 1 active RPC connection to work, do you want to reset to the default list instead?</p>
 					<button class = 'btn btn--outline' style = 'font-weight: 600' onClick = { loadDefaultRpcs }>Yes, load the default RPC list</button>
 				</aside>
 				<ul class = 'grid' style = '--gap-y: 0.5rem'>
-					<RpcSummary entries = { rpcEntries } info = { lastEntry } />
+					<RpcSummary info = { latestEntry.value } />
 				</ul>
 			</>
 		)
 	}
 
 	return (
 		<ul class = 'grid' style = '--gap-y: 0.5rem'>
-			{ rpcEntries.map(entry => <RpcSummary entries = { rpcEntries } info = { entry } />) }
+			{ rpcEntries.value.map(entry => <RpcSummary info = { entry } />) }
 		</ul>
 	)
 }
 
-const RpcSummary = ({ entries, info }: { entries: RpcEntries, info: RpcEntry }) => {
+const RpcSummary = ({ info }: { info: RpcEntry }) => {
 	const networkName = getChainName(info.chainId)
+
+	// rerender form if entry is updated in the background by specifying a unique key
+	const infoHash = Object.values(info).join('')
```

Comment:
```suggestion
	const infoKey = Object.values(info).join('|')
```
as this is not a hash, I would not call it a hash. Its also often better to use some value to join with so values of other don't conflict with each other

Reply:
added in recent commit [92daa24](https://github.com/DarkFlorist/TheInterceptor/pull/1029/commits/92daa249f367de1ff75dedc7dea55bae91fc12a4)